nix/build-maven-package.nix

```diff
@@ -0,0 +1,74 @@
+{ lib, stdenv, maven }:
+
+{ src, sourceRoot ? null, buildOffline ? false, patches ? [ ], pname, version
+, mvnHash ? "", mvnFetchExtraArgs ? { }, mvnDepsParameters ? ""
+, manualMvnArtifacts ? [ ], mvnParameters ? "", ... }@args:
+
+# originally extracted from dbeaver
+# created to allow using maven packages in the same style as rust
+
+let
+  fetchedMavenDeps = stdenv.mkDerivation ({
+    name = "${pname}-${version}-maven-deps";
+    inherit src sourceRoot patches;
+
+    nativeBuildInputs = [ maven ];
+
+    buildPhase = ''
+      runHook preBuild
+    '' + lib.optionalString buildOffline ''
+      mvn org.apache.maven.plugins:maven-dependency-plugin:3.6.1:go-offline -Dmaven.repo.local=$out/.m2 ${mvnDepsParameters}
+
+      for artifactId in ${builtins.toString manualMvnArtifacts}
+      do
+        echo "downloading manual $artifactId"
+        mvn dependency:get -Dartifact="$artifactId" -Dmaven.repo.local=$out/.m2
+      done
+    '' + lib.optionalString (!buildOffline) ''
+      mvn package -Dmaven.repo.local=$out/.m2 ${mvnParameters}
+    '' + ''
+      runHook postBuild
+    '';
+
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''
+      runHook preInstall
+
+      find $out -type f \( \
+        -name \*.lastUpdated \
+        -o -name resolver-status.properties \
+        -o -name _remote.repositories \
+        -o -name \*.snapshots.xml \
+        -o -name \*.snapshots.xml.sha1 \) \
```

Comment:
we still get `....snapshots.xml` files which contain dates of the latest snapshot, which we need to remove to make the build (hopefully) deterministic